java/sage/media/format/MPEGParser2.java

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sage.media.format;
+
+import sage.Sage;
+
+import java.io.OutputStream;
+
+public class MPEGParser2
+{
+  public static final int REMUX_TS = 1;
+  public static final int REMUX_PS = 16;
+
+  public enum StreamFormat
+  {
+    /**
+     * The stream doesn't have a specific format. Select the first channel. (This may result in
+     * audio only.)
+     */
+    UNKNOWN,
+    /**
+     * The stream is formatted as ATSC. Searches for the desired streams will be based on this.
+     * (This will always have video.)
+     */
+    ATSC,
+    /**
+     * The stream is formatted as DVB.  Searches for the desired streams will be based on this.
+     */
+    DVB,
+    /**
+     * The stream doesn't have a specific format. (This may result in audio only.)
+     */
+    FREE
+  }
+
+  public enum SubFormat
+  {
+    UNKNOWN,
+    TERRESTRIAL,
+    CABLE,
+    SATELLITE
+  }
+
+  public enum TuneStringType
+  {
+    CHANNEL,
+    PROGRAM,
+    XX_XX,
+    XX_XX_XX,
+    AUTO
+  }
+
+  static
+  {
+    if (System.getProperty("os.name").toLowerCase().contains("windows") ||
```

Comment:
You can use the WINDOWS_OS and LINUX_OS constants in Sage.java instead; it always loads first and those values are safe to use in other static initializers as well.